xmppserver/src/main/java/org/jivesoftware/database/DbConnectionManager.java

```diff
@@ -145,13 +146,16 @@ public static Connection getConnection() throws SQLException {
                         "(attempt " + currentRetryCount + " out of " + maxRetries + ").", e);
             }
             
-            try {
-                Thread.sleep(retryWait);
-            } catch (Exception e) {
-                // Ignored, the thread was interrupted while waiting, so no need to log either
-            }
             currentRetryCount++;
-        } while (currentRetryCount <= maxRetries);
+            loopIfNoConnection = currentRetryCount <= maxRetries;
+            if (loopIfNoConnection) {
+                try {
+                    Thread.sleep(retryWait);
+                } catch (Exception e) {
```

Comment:
You're catching an InterruptedException here, which really means you should re-interrupt the thread. Something like

```
try {
  Thread.sleep(retryWait);
} catch (InterruptedException ex) {
  Log.info("Interrupted whilst waiting", e);
  Thread.currentThread().interrupt(); 
}
```

Reply:
Hmmm... shouldn't we consider that an interrupt here is likely an attempt to break the retry loop completely. Maybe we should throw an SQLException?

Reply:
Fair point - I'll have to read up a little to be honest and be sure what is good practice.

Reply:
So if I'm thinking along the right lines (@GregDThomas thanks for the link) - I'm leaning towards a combination of your two suggestions - maybe something like this:

```
try {
  Thread.sleep(retryWait);
} catch (InterruptedException ex) {
  String msg = "Interrupted waiting for DB Connection";
  Log.info(msg,ex);
  Thread.currentThread().interrupt(); 
  throw new SQLException(msg,ex);
}
```
We don't just swallow the interruption, but we fail fast from the point of view of the caller, too.

Reply:
Yup, concur with that :)